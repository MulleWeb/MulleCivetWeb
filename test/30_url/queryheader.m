#import <MulleCivetWeb/MulleCivetWeb.h>


static void   test( NSDictionary *info)
{
   NSURL          *url;
   NSDictionary   *headers;
   NSString       *s;
   NSString       *query;

   s     = @"http://127.0.0.1/action";
   query = [info mulleURLEscapedQueryString];
   if( query)
      s = [NSString stringWithFormat:@"%@?%@", s, query];
   url = [NSURL URLWithString:s];

   printf( "%s\n", [url UTF8String]);

   headers = [url mulleQueryDictionary];
   if( info && ! [info isEqualToDictionary:headers])
      printf( "FAIL %s <> %s\n", [info UTF8String], [headers UTF8String]);
}


int   main( int argc, char *argv[])
{
   test( nil);
   test( @{});
   test( @{ @"a": @"b" });
   test( @{ @"=a&": @"=b&" });

   return( 0);
}
