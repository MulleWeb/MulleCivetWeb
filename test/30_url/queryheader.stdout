http://127.0.0.1/action
http://127.0.0.1/action
http://127.0.0.1/action?a=b
http://127.0.0.1/action?%3da%26=%3db%26
